lib/acme/client.rb

```diff
@@ -88,24 +93,6 @@ def connection
     end
   end
 
-  def challenge_from_hash(arguments)
-    attributes = arguments.to_h
-    %w(type uri token).each do |key|
-      raise ArgumentError, "missing key: #{key}" unless attributes.key?(key)
-    end
-
-    case attributes.fetch('type')
-    when 'http-01'
-      Acme::Client::Resources::Challenges::HTTP01.new(self, attributes)
-    when 'dns-01'
-      Acme::Client::Resources::Challenges::DNS01.new(self, attributes)
-    when 'tls-sni-01'
-      Acme::Client::Resources::Challenges::TLSSNI01.new(self, attributes)
-    else
-      raise 'Unsupported resource type'
-    end
-  end
-
```

Comment:
Just thinking it might be a good time to go for 1.0 and start taking API stability more into consideration.

The gem is pretty stable at this point I would say. I already provisioned a bit more than 1,080,000 certificates with it.